sdk/lib/string/memchr.c

```diff
@@ -0,0 +1,21 @@
+// Copyright SCI Semiconductor and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#include <string.h>
+
+const void *__cheri_libcall memchr(const void *voidString, int intChar, size_t n)
+{
+	const char  c = (char)intChar;
+	const char *s = (const char *)voidString;
```

Comment:
The specification for `memchr` requires that the comparison be interpreting the bytes as `unsigned char`, not `char`.